src/mz/tests/e2e.rs

```diff
@@ -0,0 +1,537 @@
+        let assert = cmd().arg("config").arg("list").assert().success();
+
+        let output = output_to_string(assert);
+        assert!(output.trim() == expected_command_output.trim());
+
+        // Assert `mz config set profile` + `mz config get profile output:
+        cmd()
+            .arg("config")
+            .arg("set")
+            .arg("profile")
+            .arg("random")
+            .assert()
+            .success();
+
+        let assert = cmd()
+            .arg("config")
+            .arg("get")
+            .arg("profile")
+            .assert()
+            .success();
+
+        let output = output_to_string(assert);
+        assert!(output.trim() == "random");
+
+        // Assert `mz config remove profile`
+        cmd()
+            .arg("config")
+            .arg("remove")
+            .arg("profile")
+            .assert()
+            .success();
+
+        let assert = cmd()
+            .arg("config")
+            .arg("get")
+            .arg("profile")
+            .assert()
+            .success();
+
+        let output = output_to_string(assert);
+        assert!(output.trim() == "default");
+
+        // Test - `mz profile`
+        //
+        // Assert `mz profile list`
+        let binding = output_to_string(
+            cmd()
+                .arg("profile")
+                .arg("config")
+                .arg("get")
+                .arg("app-password")
+                .assert()
+                .success(),
+        );
+        let app_password = binding.trim();
+
+        #[derive(Deserialize, Serialize, Tabled)]
+        pub struct ProfileConfigParam<'a> {
+            #[tabled(rename = "Name")]
+            name: &'a str,
+            #[tabled(rename = "Value")]
+            value: &'a str,
+        }
+
+        let vec = vec![
+            ProfileConfigParam {
+                name: "admin-endpoint",
+                value: "<unset>",
+            },
+            ProfileConfigParam {
+                name: "app-password",
+                value: app_password,
+            },
+            ProfileConfigParam {
+                name: "cloud-endpoint",
+                value: "<unset>",
+            },
+            ProfileConfigParam {
+                name: "region",
+                value: "aws/us-east-1",
+            },
+            ProfileConfigParam {
+                name: "vault",
+                value: "<unset>",
+            },
+        ];
+        let expected_command_output = Table::new(vec).with(Style::psql()).to_string();
+        let binding = cmd()
+            .arg("profile")
+            .arg("config")
+            .arg("list")
+            .assert()
+            .success();
+
+        let output = output_to_string(binding);
+        assert!(output.trim() == expected_command_output.trim());
+    }
+
+    /// TODO: Re-enable after understanding how to get an CI app-password, admin endpoint and cloud endpoint.
```

Comment:
Do we have a tracking issue for `mz` somewhere? If not, we should file one, and make sure this is listed as a blocker for public preview!